src/Microsoft.AspNetCore.Antiforgery/Internal/DefaultAntiforgeryTokenGenerator.cs

```diff
@@ -60,16 +61,17 @@ public AntiforgeryToken GenerateCookieToken()
             };
 
             var isIdentityAuthenticated = false;
-            var identity = httpContext.User?.Identity as ClaimsIdentity;
+            var authenticatedIdentity = httpContext.User?.Identities?.FirstOrDefault(c => c.IsAuthenticated == true);
```

Comment:
Also,
- Can `Identities` really be `null`? Check the real code and don't check here unless it's a real scenario.
- Lots of boilerplate around `authenticatedIdentity` repeated in two methods. Suggest a helper method.